cmd/promoted-image-governor/main.go

```diff
@@ -441,7 +441,11 @@ func main() {
 			if _, ok := opts.explains[isTagRef]; ok {
 				opts.explains[isTagRef] = cfg.Metadata.AsString()
 			}
-			if cfg.PromotionConfiguration.TagByCommit {
+			var taggedByCommit bool
+			for _, target := range api.PromotionTargets(cfg.PromotionConfiguration) {
+				taggedByCommit = taggedByCommit || target.TagByCommit
+			}
```

Comment:
This checks if _any_ tag is created by commit, not just `isTagRef`.

Reply:
Yes, I'm happy to make an issue for someone to make that better in the future but I am at the boundary of how much I can refactor here. The regex created will be a superset of the regex that's needed, and if for some reason images tagged by PR from another `Targets` stanza existed, they would not get cleaned up. The refactor here would require that `release.PromotedTags` return something other than a flat `isTagRef` list.

Reply:
We can iterate over `PromotionTargets`, which is where all the information we need is.

There would still be some ways to create an unfortunate configuration which would not be processed correctly, but that's already the case for the current implementation (before `Targets`).

Reply:
(Not impossible, just asking for this to be a follow-up that someone on DPTP could handle?)

Reply:
I see `PromotedTags` is quite messy already.  At least leave a comment here, then.